exchanges/okcoin/okcoin_wrapper.go

```diff
@@ -605,89 +594,160 @@ func (o *OKCoin) FetchAccountInfo(ctx context.Context, assetType asset.Item) (ac
+			TransferType:    "withdrawal",
 		}
 	}
 	return resp, nil
 }
 
 // SubmitOrder submits a new order
 func (o *OKCoin) SubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	if s == nil {
+		return nil, fmt.Errorf("%w, place order request parameter can not be null", common.ErrNilPointer)
+	}
+	if s.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w, asset: %v", asset.ErrNotSupported, s.AssetType)
+	}
 	err := s.Validate()
 	if err != nil {
 		return nil, err
 	}
-
-	fPair, err := o.FormatExchangeCurrency(s.Pair, s.AssetType)
+	s.Pair, err = o.FormatExchangeCurrency(s.Pair, s.AssetType)
 	if err != nil {
 		return nil, err
 	}
-
-	req := PlaceOrderRequest{
-		ClientOID:    s.ClientID,
-		InstrumentID: fPair.String(),
-		Side:         s.Side.Lower(),
-		Type:         s.Type.Lower(),
-		Size:         strconv.FormatFloat(s.Amount, 'f', -1, 64),
-	}
-	if s.Type == order.Limit {
-		req.Price = strconv.FormatFloat(s.Price, 'f', -1, 64)
+	if s.TradeMode == "" {
+		s.TradeMode = "cash"
+	}
+	req := PlaceTradeOrderParam{
+		ClientOrderID: s.ClientID,
+		InstrumentID:  s.Pair,
+		Side:          s.Side.Lower(),
+		OrderType:     s.Type.Lower(),
+		Size:          s.Amount,
+		TradeMode:     s.TradeMode,
+		Price:         s.Price,
+		OrderTag:      "",
+		BanAmend:      false,
+	}
+	if (s.Type == order.Limit || s.Type == order.PostOnly ||
+		s.Type == order.FillOrKill || s.Type == order.ImmediateOrCancel) && s.Price <= 0 {
+		return nil, fmt.Errorf("%w, price is required for order types %v,%v,%v, and %v", errInvalidPrice, order.Limit, order.PostOnly, order.ImmediateOrCancel, order.FillOrKill)
+	}
+	var orderResponse *TradeOrderResponse
+	if o.Websocket.IsConnected() && o.Websocket.CanUseAuthenticatedEndpoints() && o.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+		orderResponse, err = o.WsPlaceOrder(&req)
+	} else {
+		orderResponse, err = o.PlaceOrder(ctx, &req)
 	}
-
-	orderResponse, err := o.PlaceSpotOrder(ctx, &req)
 	if err != nil {
 		return nil, err
 	}
-
-	if !orderResponse.Result {
-		return nil, order.ErrUnableToPlaceOrder
-	}
 	return s.DeriveSubmitResponse(orderResponse.OrderID)
 }
 
 // ModifyOrder will allow of changing orderbook placement and limit to
 // market conversion
-func (o *OKCoin) ModifyOrder(_ context.Context, _ *order.Modify) (*order.ModifyResponse, error) {
-	return nil, common.ErrFunctionNotSupported
+func (o *OKCoin) ModifyOrder(ctx context.Context, req *order.Modify) (*order.ModifyResponse, error) {
+	if req == nil {
+		return nil, fmt.Errorf("%w, modify request parameter can not be null", common.ErrNilPointer)
+	}
+	if req.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w, asset: %v", asset.ErrNotSupported, req.AssetType)
+	}
+	var err error
+	req.Pair, err = o.FormatExchangeCurrency(req.Pair, asset.Spot)
+	if err != nil {
+		return nil, err
+	}
+	err = req.Validate()
+	if err != nil {
+		return nil, err
+	}
+	request := &AmendTradeOrderRequestParam{
```

Comment:
rename request as it collides with a package name

Please do this across this file

Reply:
Thanks for that. There are two more inside of the websocket file:
<img width="1073" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9261323/4f8333ab-55bc-41e6-8505-fd47688434c3">